sdk/timeseriesinsights/Instance_API_Design.md

```diff
@@ -0,0 +1,64 @@
+# Time Series Insights
+
+## Instances
+Time Series Model instances are virtual representations of the time series themselves. Instances have descriptive information associated with them called instance properties, such as a time series ID, type, name, description, hierarchies, and instance fields. At a minimum, instance properties include hierarchy information.
+
+### GET /timeseries/instances
+
+```csharp
+public virtual Pageable<TimeSeriesInstance> GetInstances(CancellationToken cancellationToken = default);
+```
+
+### POST /timeseries/instances/suggest
+
+```csharp
+public virtual Response<InstancesSearchStringSuggestion[]> GetSearchSuggestions(InstancesSuggestRequest instancesSuggestRequest, CancellationToken cancellationToken = default);
+
+// Thoughts on renaming InstancesSuggestRequest to GetSearchSuggestionsOptions
+// Thoughts on renaming InstancesSearchStringSuggestion to InstanceSuggestion
+```
+
+### POST /timeseries/instances/$batch
+
+```csharp
+public virtual Response<InstanceOrError[]> GetInstances(TimeSeriesId[] timeSeriesIds, CancellationToken cancellationToken = default);
+
+// Thoughts on renaming InstanceOrError to GetInstanceOperationResult
+```
+
+```csharp
+public virtual Response<InstanceOrError[]> GetInstances(string[] timeSeriesNames, CancellationToken cancellationToken = default);
+
+// Thoughts on naming the APIs GetInstancesByIds and GetInstancesByNames
+```
+
+```csharp
+public virtual Response<InstanceOrError[]> CreateOrReplaceInstances(TimeSeriesInstance[] timeSeriesInstances, CancellationToken cancellationToken = default);
+```
+
+```csharp
+public virtual Response<InstanceOrError[]> ReplaceInstances(TimeSeriesInstance[] timeSeriesInstances, CancellationToken cancellationToken = default);
+
+// thoughts on only having CreateOrReplaceInstances? It's safer to have both. 
```

Comment:
This is what I see for guidance:
https://azure.github.io/azure-sdk/dotnet_introduction.html#naming

It mentions set as a standard verb for "Creates or replaces a resource."

Reply:
Thinking about this more, one is Put, one is Update in which case separate API seems a better option.

`/// <summary> Time series instances to be created or updated. </summary>
        public IList<TimeSeriesInstance> Put { get; }
        /// <summary> Time series instance to be updated. If instance does not exist, an error is returned. </summary>
        public IList<TimeSeriesInstance> Update { get; }`

Reply:
Sure. Let's sync offline regarding this one.